xrtpy/response/temperature_response.py

```diff
@@ -19,43 +21,160 @@
+
+
 class TemperatureResponseFundamental:
     """Produce the temperature response for each XRT x-ray channel, assuming a spectral emission model."""
 
-    def __init__(self, filter_name, observation_date):
+    def __init__(self, filter_name, observation_date, abundance_model):
         self._name = resolve_filter_name(filter_name)
         self.observation_date = observation_date
         self._channel = Channel(self.name)
+        self._abundance_model = resolve_abundance_model_type(abundance_model)
+
+    @property
+    def abundances(self) -> Dict[str, Real]:
+        return self._abundances
+
+    """
+    @abundances.setter
+    def abundances(self, abundance_model: str):
+        if abundance_model == "chianti":
+            print('abundance type: ', abundance_model)
+        else:
+            print('abundance type: ', abundance_model)
+
+
+    @abundances.setter
+    def abundances(self, abundance_model: str):
+        if abundance_model == "coronal":
+            self._abundances = {
+                "logged_temperature": _XRT_coronal_chianti_emiss_model["LOGTE"],
+                "wavelength": _XRT_coronal_chianti_emiss_model["LMBDA"],
+                "corona_solar_spectra": _XRT_coronal_chianti_emiss_model["SOLSPEC"],
+                "spectra_generated_information": _XRT_coronal_chianti_emiss_model[
+                    "HEADER"
+                ]["TEXT"],
+            }
+        elif abundance_model == "photospheric":
+            self._abundances = {"C": 1.4e-3}
+    """
+
+    @property
+    def abundance_model(self):
+        """Name of abundance model."""
+
+        return self._abundance_model
 
     @property
     def name(self):
         """Name of searched filter."""
         return self._name
 
+    @property
+    def get_abundance_data(self):
+        abundance_type = self.abundance_model
+        if abundance_type == "chianti":
+            data = _abundance_model_data[self.abundance_model]
+            return {
+                "CHIANTI_abundance_model": data["ABUND_MODEL"][0],
+                "dens_model": data["DENS_MODEL"][0],
+                "ioneq_model": data["IONEQ_MODEL"][0],
+                "name": data["NAME"][0],
+                "spectra": data["SPEC"][0],
+                "spectra_units": data["SPEC_UNITS"][0],
+                "temperature": data["TEMP"][0],
+                "temp_units": data["TEMP_UNITS"][0],
+                "tlength": data["TLENGTH"][0],
+                "wlength": data["WLENGTH"][0],
+                "wavelength": data["WAVE"][0],
+                "wavelength_units": data["WAVE_UNITS"][0],
+            }
+        else:
+            data = _abundance_model_data[self.abundance_model]
+            return {
+                "temperature": data["LOGTE"],
+                "wavelength": data["LMBDA"],
+                "spectra": data["SOLSPEC"],
+                "header_information": data["HEADER"]["TEXT"],
+            }
+
+    @property
+    @u.quantity_input
```

Comment:
```suggestion
    @u.quantity_input
    @property
```
This might have a non-zero chance of working?